src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1170,48 +1174,56 @@ public static void Reverse(ref byte buf, nuint length)
                     tempLast = Avx2.Permute2x128(tempLast, tempLast, 0b00_01);
 
                     // Store the reversed vectors
-                    tempLast.StoreUnsafe(ref buf, firstOffset);
-                    tempFirst.StoreUnsafe(ref buf, lastOffset);
-                }
-                buf = ref Unsafe.Add(ref buf, numIters * numElements);
-                length -= numIters * numElements * 2;
+                    tempLast.StoreUnsafe(ref first);
+                    tempFirst.StoreUnsafe(ref last);
+
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Subtract(ref last, Vector256<byte>.Count);
```

Comment:
The overall handling of mutating `byrefs` is potentially buggy and error prone. It's really easy to accidentally do the wrong thing, especially over the course of several refactorings.

I think it's ok/better to leave some small perf on the table in order to keep the code more readable/maintainable and less error prone.

CC. @jkotas

Reply:
In this particular case, it would be good to look at where the codegen deficiencies between the two approaches are.

We could likely declare `firstOffset`/`lastOffset` outside the loop and do simple increments to improve codegen a bit, for example.

We should also validate that `numElements` is being treated as a constant by the JIT. If not, we should inline the usages of `Vector256<byte>.Count` or use the existing internal constant `Vector256.Size`.

Reply:
I'll take a look at this on Friday. Should be a simple fix to use offsets instead

Reply:
Changed to offsets, is it better now?